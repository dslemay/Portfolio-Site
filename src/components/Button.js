import React from 'react';
import PropTypes from 'prop-types';
import Link from 'gatsby-link';
import styled from 'react-emotion';
import { css } from 'emotion';

const ButtonStyles = props => css`
  display: block;
  background-color: ${props.theme.complementaryLight};
  color: ${props.theme.baseColor};
  padding: 0.2rem 1.3rem;
  border-radius: 20px;
  text-decoration: none;
  cursor: pointer;
`;

const AnchorTag = styled.a`
  ${ButtonStyles};
`;

const LinkTag = styled(Link)`
  ${ButtonStyles};
`;

const BasicButton = styled.button`
  ${ButtonStyles};
`;

const Button = props => {
  if (props.href) {
    return (
      <AnchorTag href={props.href} target="_blank">
        {props.children}
      </AnchorTag>
    );
  }

  if (props.to) {
    return <LinkTag to={props.to}>{props.children}</LinkTag>;
  }

  return <BasicButton type={props.type}>{props.children}</BasicButton>;
};

Button.defaultProps = {
  type: 'button'
};

Button.propTypes = {
  href: PropTypes.string,
  to: PropTypes.string,
  type: PropTypes.oneOf(['button', 'submit'])
};

export default Button;
