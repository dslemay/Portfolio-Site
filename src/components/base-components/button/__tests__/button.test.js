import React from 'react';
import { cleanup, render } from 'react-testing-library';

import Button from '..';

describe('<Button />', () => {
  afterEach(cleanup);

  it('renders a button correctly', () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an internal link correctly', () => {
    const { container } = render(<Button url="/about">About</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders an anchor link correctly', () => {
    const { container } = render(<Button url="https://github.com/dlsemay">My Github</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
