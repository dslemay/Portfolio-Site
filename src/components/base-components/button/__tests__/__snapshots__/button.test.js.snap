// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a button correctly 1`] = `
<button
  class="css-lwkiyc"
  type="button"
>
  Click me
</button>
`;

exports[`<Button /> renders an anchor link correctly 1`] = `
<a
  href="https://github.com/dlsemay"
  rel="noopener noreferrer"
  target="_blank"
>
  My Github
</a>
`;

exports[`<Button /> renders an internal link correctly 1`] = `
<mockedlink
  class="css-lwkiyc"
  to="/about"
>
  About
</mockedlink>
`;
